Library/ViewModels/ChangeEmailViewModelTests.swift

```diff
@@ -179,19 +184,102 @@ final class ChangeEmailViewModelTests: TestCase {
   }
 
   func testResendVerificationStackViewIsHidden_IfEmailIsVerified() {
-    self.resendVerificationStackViewIsHidden.assertDidNotEmitValue()
+    self.vm.inputs.viewDidLoad()
+
+    self.scheduler.advance()
+
+    self.resendVerificationEmailViewIsHiddenObserver
+      .assertValues([true], "Email is deliverable and verified")
   }
 
   func testResendVerificationStackViewIsNotHidden_IfEmailIsNotVerified() {
-    self.resendVerificationStackViewIsHidden.assertDidNotEmitValue()
+    let userEmailFields = UserEmailFields.template
+      |> \.isEmailVerified .~ false
+
+    let mockService = MockService(changeEmailResponse: UserEnvelope(me: userEmailFields))
+
+    withEnvironment(apiService: mockService) {
+      self.vm.inputs.viewDidLoad()
+
+      self.scheduler.advance()
+
+      self.resendVerificationEmailViewIsHiddenObserver.assertValues([false], "Email is unverified")
+    }
   }
 
-  func testResendVerificationStackViewAppears_AfterChangingEmail() {
+  func testResendVerificationStackViewIsNotHidden_IfEmailIsUndeliverable() {
+    let userEmailFields = UserEmailFields.template
+      |> \.isDeliverable .~ false
+
+    let mockService = MockService(changeEmailResponse: UserEnvelope(me: userEmailFields))
+
+    withEnvironment(apiService: mockService) {
+      self.vm.inputs.viewDidLoad()
+
+      self.scheduler.advance()
+
+      self.resendVerificationEmailViewIsHiddenObserver
+        .assertValues([false], "Email is undeliverable")
+    }
+  }
+
+  func testWarningMessageLabel_isHidden() {
```

Comment:
Here I would make the test name a little more descriptive, like: `testWarningMessageLabel_isHidden_If...`